### DISCUSSION TITLE
Get Host Path Feature

### DISCUSSION THREAD

**Discussion by Mikenux** [flatpak/xdg-desktop-portal#1433] [General]
This feature has been implemented in the main branch. This means, if I'm not wrong, that it will only be available for version 1.20. If so, doesn't this mean that apps will still have to rely on the `host` or `home` filesystem permission to display the host path on versions prior to 1.20?

**Comment by smcv** [comment]
What else do you imagine would happen? We don't have a time machine, so older versions of xdg-desktop-portal are not going to implement newer features.

**Reply by Mikenux** [reply]
What would happen is that apps would still use `filesystem=host/home` to display paths if all systems don't have the latest version of xdg-desktop-portal. Or is there something I missed?

**Comment by GeorgesStavracas** [comment]
What you're describing (or complaining, I'm not sure) is the concept of "stable releases", and "frozen in time distros".

There is nothing we can do about it. We cannot retroactively add new features to stable versions as that would break the stability rules we follow. We also cannot go back in time to release stable versions with the features, for obvious reasons.

All there is to do is to ask for distributors to update to the latest release. If they decide not to, that's a problem between the distribution and their users, they have to handle that themselves.

**Reply by Mikenux** [reply]
No, I'm not complaining. I just have a concern.

The problem is that app developers cannot move away from `filesystem=home` or `filesystem=host` for distributions that will not include version 1.20. Consequently, as I understand it, this also means that they can't move away from this permission even on distributions that ship with 1.20.

Note that you support, according to https://github.com/flatpak/xdg-desktop-portal/security, more than just the latest version. Therefore, upgrading to the latest version is not mandatory.

A solution is needed, and I know that solution isn't backporting new features. This solution consists of either having (for app developers) something that is compatible (i.e. use of `filesystem` or `gethostpath` depending on what is supported), or simply letting time pass if the time to support the versions prior to 1.20 is not too long (also taking into account the time it takes for app developers to adapt their apps to display the path using the new method).

**Comment by swick** [comment]
I think @Mikenux has a fair point. This is another case where a fallback mechanism (e.g. https://github.com/flatpak/flatpak/pull/5852) might be helpful.

It's however not entirely clear to me that there are apps which require `--filesystem=something` only for being able to get the host path. Can you point out any where that's the case?

**Reply by Mikenux** [reply]
Unfortunately, not so much. Maybe KeePassXC? I had a quick look at its documentation and there is entering database URLs (but developers can replace this with file selection; but it's up to them). Otherwise, other cases I know also use neighboring files (for example).

**Reply by swick** [reply]
In that case I don't think we should invest any more energy on this issue.

**Reply by Mikenux** [reply]
Seems fine. I guess neighboring files won't be coming out anytime soon.
